### PR TITLE
feat(particle-data)!: use MsgPack for particle data serialization [Fixes VM-407]

### PR DESCRIPTION
**Pull request by monoid**
## Description
Use multicodec-prefixed MsgPack for particle data serialization.

## Motivation
MsgPack is more efficient format by memory, CPU and transmitted data size than JSON that was used before.  The multicodec allows to change formats in incompatibility-detectable way and, in the future, support multiple formats.

## Related Issue(s)
[VM-407](https://linear.app/fluence/issue/VM-407/use-msgpack-for-nox-particle-data)

## Proposed Changes
Instead of `asynchronous_codec::JsonCode` which is not used in a synchronous context anyway, use AquaVM sede crate with multiformats support.  The outer `UveCodec` is left as is.

## Checklist
- [🤔] The code follows the project's coding conventions and style guidelines.
- [x] All tests related to the changes have passed successfully.
- [x] Documentation has been updated to reflect the changes (if applicable).
- [x] All new and existing unit tests have passed.
- [x] I have self-reviewed my code and ensured its quality.
- [x] I have added/updated necessary comments to aid understanding.

## Reviewer Checklist
- [ ] Code has been reviewed for quality and adherence to [guidelines](https://doc.rust-lang.org/1.0.0/style/README.html).
- [ ] Tests have been reviewed and are sufficient to validate the changes.
- [ ] Documentation has been reviewed and is up to date.
- [ ] Any questions or concerns have been addressed.

### CODE REVIEW

File: connection-pool/src/behaviour.rs [hunk 1]
```diff
@@ -635,7 +635,6 @@ impl NetworkBehaviour for ConnectionPoolBehaviour {
                     .push_back(ExtendedParticle::new(particle, root_span));
                 self.wake();
             }
-            Ok(HandlerMessage::InboundUpgradeError(err)) => log::warn!("UpgradeError: {:?}", err),
```
Comment (gurinderu) [inline connection-pool/src/behaviour.rs:638]: why was this removed?

Comment (gurinderu) [reply connection-pool/src/behaviour.rs:638]: Got it

Comment (monoid) [reply connection-pool/src/behaviour.rs:638]: It is not used.

**Review by gurinderu — APPROVED** [review]

File: particle-protocol/Cargo.toml [hunk 2]
```diff
@@ -21,9 +21,11 @@ log = { workspace = true }
 derivative = { workspace = true }
 base64 = { workspace = true }
 tokio = { workspace = true, features = ["sync"] }
-asynchronous-codec = { version = "0.7.0", features = ["json"] }
+asynchronous-codec = { version = "0.7.0" }
 unsigned-varint = { version = "0.8.0", features = ["codec", "asynchronous_codec"] }
 tracing = { workspace = true }
+air-interpreter-sede = { version = "0.1.0", features = ["msgpack"] }
```
Comment (folex) [inline particle-protocol/Cargo.toml:27]: Could you please also update it in js-client? Create a PR there, and we will use it in e2e

Comment (monoid) [reply particle-protocol/Cargo.toml:27]: Yes, that's what I'm working on right now.

File: particle-protocol/src/libp2p_protocol/codec/fluence.rs [hunk 3]
```diff
@@ -1,21 +1,32 @@
 use crate::ProtocolMessage;
-use asynchronous_codec::{BytesMut, Decoder, Encoder, JsonCodec, JsonCodecError};
+use air_interpreter_sede::{
```
[the remark this answers is not in the document]
Comment (monoid) [reply particle-protocol/src/libp2p_protocol/codec/fluence.rs:2]: It doesn't, it merely provides infrastructure.  The format is defined by `ProtocolMessageFormat` definition below and can be changed literally with one line change.

[the remark this answers is not in the document]
Comment (folex) [reply particle-protocol/src/libp2p_protocol/codec/fluence.rs:2]: Got it, makes sense.
